core/src/test/scala/unit/kafka/log/LogValidatorTest.scala

```diff
@@ -488,8 +496,11 @@ class LogValidatorTest {
     }
     assertEquals(now + 1, validatingResults.maxTimestampMs,
       s"Max timestamp should be ${now + 1}")
-    assertEquals(2, validatingResults.shallowOffsetOfMaxTimestamp,
-      "Shallow offset of max timestamp should be 2")
+
+    // Both V2 and V1 has single branch in the record when compression is enable, and hence their shallow OffsetOfMaxTimestamp
+    // is the last offset of the single branch
+    assertEquals(1, records.batches().asScala.size)
```

Comment:
there is a minor issue with "checkRecompression":
1. at line 455, `records` should be created with "CompressionType.NONE".
2. at line 502, the check should be `assertEquals(1, validatedRecords.batches().asScala.size)`

Reply:
@vincent81jiang : Thanks for the comment. The line at 455 seems to be wrong. What do you think @chia7712 ?

Reply:
@vincent81jiang you are totally right. I have filed PR (https://github.com/apache/kafka/pull/15948) to fix that